test/box-luatest/gh_8051_set_box_cfg_thru_env_test.lua

```diff
@@ -0,0 +1,93 @@
+local server = require('luatest.server')
+local popen = require('popen')
+local clock = require('clock')
+
+local t = require('luatest')
+local g = t.group()
+
+g.after_all = function()
+    if g.server ~= nil then
+       g.server:stop()
+   end
+end
+
+local TARANTOOL_PATH = arg[-1]
+
+local function popen_run(command)
+    local cmd = TARANTOOL_PATH .. ' -i 2>&1'
+    local ph = popen.new({cmd}, {
+        shell = true,
+        setsid = true,
+        group_signal = true,
+        stdout = popen.opts.PIPE,
+        stderr = popen.opts.PIPE,
+        stdin = popen.opts.PIPE,
+    })
+    t.assert(ph, 'process is not up')
+
+    ph:write(command)
+
+    local output = '';
+    local time_quota = 2.0
+    local start_time = clock.monotonic();
+    while output:find('message') == nil
+        and clock.monotonic() - start_time < time_quota do
+        local data = ph:read({timeout = 1.0})
+        if data ~= nil then
+            output = output .. data
+        end
+    end
+
+    ph:close()
+    return output
+end
+
+g.test_json_table_curly_bracket = function()
+    local env = {["TT_METRICS"] = '{"labels":{"alias":"gh_8051"},' ..
+                                  '"include":"all","exclude":["vinyl"]}'}
+
+    g.server = server:new{alias='json_table_curly_bracket', env=env}
+    g.server:start()
+
+    t.assert_equals(g.server:get_box_cfg().metrics.labels.alias, 'gh_8051')
+end
+
+g.test_json_table_square_bracket = function()
+    local res = popen_run([=[
+        os.setenv('TT_LISTEN', '["localhost:9999"]')
+        box.cfg{}
```

Comment:
Here we can get 'address already in use'. Use a Unix socket or binding to zero port.

Reply:
Done.